### PR TITLE
Refactors all our network messages

### PR DESCRIPTION
**Pull request by Asherslab**
# Changes proposed in this pull request:
- Finally organise all our damn network messages
- Removed a bunch of boiler plate to reduce possible errors in the future

Review please

### CODE REVIEW

**Review by someaddons — DISMISSED** [review]

**Comment by Asherslab** [comment]
@someaddons dammit, i'm blaming you if it breaks something because you didn't review it properly =P

**Comment by someaddons** [comment]
I'm not going to read 10k lines where 99% is repasted! Approved and I'll blame you if sth goes wrong =)

File: src/main/java/com/minecolonies/coremod/network/messages/server/colony/OpenInventoryMessage.java [hunk 1]
```diff
@@ -175,30 +146,17 @@ private void doCitizenInventory(final ServerPlayerEntity player)
...
-            @NotNull final ScarecrowTileEntity scarecrowTileEntity = (ScarecrowTileEntity) BlockPosUtil.getTileEntity(CompatibilityUtils.getWorldFromEntity(player), tePos);
-            NetworkHooks.openGui(player, scarecrowTileEntity);
-
-        }
-    }
-
-    private static boolean checkPermissions(final IColony colony, final ServerPlayerEntity player)
-    {
-        //Verify player has permission to change this huts settings
-        return colony.getPermissions().hasPermission(player, Action.MANAGE_HUTS);
```
Comment (ToMe25) [inline src/main/java/com/minecolonies/coremod/network/messages/server/colony/OpenInventoryMessage.java:201]: does this mean the permission doesn't get checked at all in this packet?

Comment (Asherslab) [reply src/main/java/com/minecolonies/coremod/network/messages/server/colony/OpenInventoryMessage.java:201]: look at the base class: `AbstractColonyServerMessage`

Comment (Asherslab) [reply src/main/java/com/minecolonies/coremod/network/messages/server/colony/OpenInventoryMessage.java:201]: that's a major part of the boiler plate i removed

Comment (ToMe25) [reply src/main/java/com/minecolonies/coremod/network/messages/server/colony/OpenInventoryMessage.java:201]: Seems ok then.
I don't know too much about minecraft/minecolonies networking, so i just looked for changes in logic that seemed important.

**Review by ToMe25 — DISMISSED** [review]
I don't know very much about this kind of networking,
but it seems to me like there are no problematic logic changes.

**Review by ToMe25 — APPROVED** [review]
As the latest commit only adds javadoc, and i am too lazy to read 10k lines of changes again, just for potenitally missing javadoc i am going to say everything is fine.

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/4850/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

**Review by marchermans — APPROVED** [review]
I am not going to read 10k line fixes. But this _looks_ fine...

**Comment by Asherslab** [comment]
Well, that's 4 approvals

**Pull request merged**